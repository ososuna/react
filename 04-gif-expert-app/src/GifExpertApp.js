
import React, { useState } from 'react';
import PropTypes from 'prop-types';

const GifExpertApp = () => {

    // const categories = ['Zenitsu', 'Sana Twice', 'Kurapika'];

    const [categories, setCategories] = useState(['Zenitsu', 'Sana Twice', 'Kurapika']);

    const handleAdd = () => {
        setCategories( ['Hunter x Hunter', ...categories] );
    }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr/>

            <button onClick = { handleAdd }>Agregar</button>

            <ol>
                {
                    categories.map( ( category ) => {
                        return <li key = { category }>{ category }</li>
                    })
                }
            </ol>

        </>
    );

}

export default GifExpertApp;
