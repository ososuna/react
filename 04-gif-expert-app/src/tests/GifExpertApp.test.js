
import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifExpertApp from '../GifExpertApp';

describe('GifExpertApp component tests', () => {
    
    test('should match with snapshot', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show a categories list', () => {
        
        const categories = ['Hunter x Hunter', 'Sana'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });
    

});
