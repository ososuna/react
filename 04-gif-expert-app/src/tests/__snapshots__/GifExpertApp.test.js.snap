// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp component tests should match with snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpertApp component tests should show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hunter x Hunter"
      key="Hunter x Hunter"
    />
    <GifGrid
      category="Sana"
      key="Sana"
    />
  </ol>
</Fragment>
`;
